contracts/modules/capital/MCR.sol

```diff
@@ -112,11 +72,28 @@ contract MCR is Iupgradable {
   function changeDependentContractAddress() public {
     qd = QuotationData(ms.getLatestAddress("QD"));
     pool = Pool(ms.getLatestAddress("P1"));
-    pd = PoolData(ms.getLatestAddress("PD"));
-    tk = NXMToken(ms.tokenAddress());
-    mr = MemberRoles(ms.getLatestAddress("MR"));
-    td = TokenData(ms.getLatestAddress("TD"));
-    proposalCategory = ProposalCategory(ms.getLatestAddress("PC"));
+    initialize();
+  }
+
+  function initialize() internal {
+
+    address currentMCR = ms.getLatestAddress("MC");
+
+    if (address(previousMCR) == address(0) || currentMCR != address(this)) {
+      // already initialized or not ready for initialization
+      return;
+    }
+
+    // fetch MCR parameters from previous contract
+    mcrFloor = uint112(previousMCR.variableMincap());
```

Comment:
proposal:

set mcrFloor to be previousMCR.getLastMCREther()

since currently in prod there is a discrepancy between the 2. if this code is launched as is, the mcr will gradually decrease to the mcrFloor (Which currently doesn't happen as part of the MCR posting)